### PR TITLE
add sign math function

### PR DESCRIPTION
**Pull request by ucasfl**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Add `sign` math function.

### CODE REVIEW

**Comment by kitaisreal** [comment]
@ucasFL could you please add documentation to this `sign` function ?

File: src/Functions/sign.cpp [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t input_rows_count) const override
+    {
+        auto zero_column = arguments[0].type->createColumnConstWithDefaultValue(input_rows_count);
+
+        ColumnWithTypeAndName left{arguments[0].column, arguments[0].type, {}};
+        ColumnWithTypeAndName right{zero_column, arguments[0].type, {}};
+
+        auto greater = FunctionFactory::instance().get("greater", context);
+        auto greater_compare = greater->build(ColumnsWithTypeAndName{left, right});
```
Comment (kitaisreal) [inline src/Functions/sign.cpp:68]: @ucasFL  maybe it make sense to extract that in variable without creating it for each function `execute` in next lines ?

**Review by kitaisreal — COMMENTED** [review]
Looks good to me.

**Comment by ucasFL** [comment]
> @ucasFL could you please add documentation to this `sign` function ?

Ok.

File: src/Functions/sign.cpp (hunk 1 above)
Comment (ucasFL) [reply src/Functions/sign.cpp]: Done!

**Comment by kitaisreal** [comment]
@ucasFL after small discussion maybe it will be better to implement it using existing framework FunctionUnaryArithmetic?
Please check for example implementation of abs.cpp.
Also could you please add tests for this `sign` function ?

**Comment by ucasFL** [comment]
> @ucasFL after small discussion maybe it will be better to implement it using existing framework FunctionUnaryArithmetic?
> Please check for example implementation of abs.cpp.
> Also could you please add performance tests for this `sign` function ?

Ok, I have rewrite it with FunctionUnaryArithmetic.

**Comment by ucasFL** [comment]
```
 01508_query_obfuscator:                                                 [ FAIL ] - result differs with reference:
2021-01-26 15:04:00 --- /ClickHouse/tests/queries/0_stateless/01508_query_obfuscator.reference	2021-01-26 14:36:59.361547309 +0300
2021-01-26 15:04:00 +++ /ClickHouse/tests/queries/0_stateless/01508_query_obfuscator.stdout	2021-01-26 15:04:00.586005177 +0300
2021-01-26 15:04:00 @@ -12,5 +12,5 @@
2021-01-26 15:04:00      AND MasonryID = 30750384
2021-01-26 15:04:00      AND intHash32(EyeballID) = 448362928 AND intHash64(EyeballID) = 12572659331310383983
2021-01-26 15:04:00      AND EarthquakeID IN (8195672321757027078, 7079643623150622129, 5057006826979676478, 7886875230160484653, 7494974311229040743)
2021-01-26 15:04:00 -    AND Photography = 1
2021-01-26 15:04:00 +    AND Sign = 1
```

I checked that obfuscator can't obfuscate reserve words like function name, so I made a little change on this test.

**Review by kitaisreal — COMMENTED** [review]
@ucasFL looks good to me, lets wait for CI and merge.

**Pull request merged**